src/allocators/HybridAllocator.sol

```diff
@@ -187,58 +192,126 @@ contract HybridAllocator is IHybridAllocator {
         return (claimHash, registeredAmounts, nonce);
     }
 
-    /// @inheritdoc IHybridAllocator
+    /// @inheritdoc IOnChainAllocation
     function permit2Allocation(
         address arbiter,
         address depositor,
         uint256 expires,
         ISignatureTransfer.TokenPermissions[] calldata permitted,
+        uint256[] calldata additionalCommitmentAmounts,
         DepositDetails calldata details,
         bytes32 claimHash,
         string calldata witness,
         bytes32 witnessHash,
-        bytes calldata signature
-    ) external returns (Lock[] memory commitments) {
-        commitments = AL.permit2Allocation(
-            arbiter, depositor, expires, permitted, details, claimHash, witness, witnessHash, signature
+        bytes calldata signature,
+        bytes calldata context // allocator signature
+    ) external returns (Lock[] memory) {
+        (Lock[] memory commitments,, bool containsAdditionalCommitments) = AL.permit2Allocation(
+            arbiter,
+            depositor,
+            expires,
+            permitted,
+            additionalCommitmentAmounts,
+            details,
+            claimHash,
+            witness,
+            witnessHash,
+            signature
         );
 
+        if (containsAdditionalCommitments) {
+            // Validate the allocator's signature for the additional commitments
+            _validateContext(commitments, additionalCommitmentAmounts, claimHash, context);
+        }
+
         // Allocate the claim
         claims[claimHash] = true;
 
         emit Allocated(depositor, commitments, details.nonce, expires, claimHash);
+
+        return commitments;
     }
 
     /// @inheritdoc IOnChainAllocation
     function prepareAllocation(
         address recipient,
         uint256[2][] calldata idsAndAmounts,
+        uint256[] calldata additionalCommitmentAmounts,
         address arbiter,
         uint256 expires,
         bytes32 typehash,
         bytes32 witness,
-        bytes calldata /* orderData */
+        bytes calldata context
     ) external returns (uint256 nonce) {
-        uint88 nonce88 = nonces + 1;
-
-        nonce =
-            AL.prepareAllocation(nonce88, recipient, idsAndAmounts, arbiter, expires, typehash, witness, ALLOCATOR_ID);
+        if (context.length > 0) {
+            // Potential off chain nonce provided
+            HybridAllocationContext calldata allocationContext = _decodeContext(context);
```

Comment:
not sure whether or not this is going to work; do we have any tests of this functionality yet? the decode-bytes-to-calldata logic is always a little gnarly/counterintuitive